src/main/java/com/aws/greengrass/shadowmanager/sync/model/FullShadowSyncRequest.java

```diff
@@ -68,8 +69,10 @@ public boolean isUpdateNecessary(SyncContext context) {
      * @throws InterruptedException     if the thread is interrupted while syncing shadow with cloud.
      */
     @Override
-    public void execute(SyncContext context) throws RetryableException, SkipSyncRequestException, InterruptedException {
+    public void execute(SyncContext context) throws RetryableException, SkipSyncRequestException,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 67 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 82-196 into a separate method.